tests/utils_test.py

```diff
@@ -350,20 +355,20 @@ def package_to_dict(cls, package):
             ('description', package.description),
             ('versionInfo', package.version),
             ('sourceInfo', package.source_info),
-            ('downloadLocation', package.download_location),
+            ('downloadLocation', str(package.download_location)),
             ('homepage', package.homepage),
             ('originator', cls.entity_to_dict(package.originator)),
             ('supplier', cls.entity_to_dict(package.supplier)),
             ('licenseConcluded', cls.license_to_dict(package.conc_lics)),
             ('licenseDeclared', cls.license_to_dict(package.license_declared)),
-            ('copyrightText', package.cr_text),
+            ('copyrightText', str(package.cr_text)),
```

Comment:
Are these additional `str` used to handle `NONE` or `NO_ASSERTION`? If so, aren't there other fields that need the same treatment, like `homepage`?

Reply:
Maybe. These are the ones that I saw in some cases. I have to admit, I didn't fully understand the issue here. It showed some differences where `NOASSERTION` was inside quotation marks on one side, and without quotation marks on the other. I debugged and noticed that in the latter case, e.g. `package.cr_text` was a `NoAssert` instance. However, its `str` and `repr` methods do return a string, so I am not quite sure why casting it explicitly as string made a slight difference.

That said, all of this is test utility code that will probably be replaced, so I was fine with just fixing the issues that I observed in any halfway-reasonable way.

Reply:
fair enough